src/Illuminate/Support/Arr.php

```diff
@@ -237,7 +237,7 @@ public static function get($array, $key, $default = null)
 
 		foreach (explode('.', $key) as $segment)
```

Comment:
Ok.  I reverted everything back to use explode and just remove the ! is_array() check.  All tests is passed and the null case is also passed.

Here's the benchmark:
![New Benchmark](http://i.imgur.com/x8DX2TT.png)